snap/snapcraft.yaml

```diff
@@ -100,25 +100,28 @@ parts:
   rockcraft:
     source: .
     plugin: python
-    python-packages:
-        - wheel
-        - pip
-        - setuptools
     python-requirements:
-        - requirements-jammy.txt
-        - requirements.txt
+        # This will be generated by override-build
+        - uv-requirements.txt
+        # This has the source for python-apt for noble
+        - requirements-noble.txt
     build-attributes:
         - enable-patchelf
     build-packages:
         - cargo
+    build-snaps:
+        - astral-uv
     build-environment:
-        - "CFLAGS": "$(pkg-config python-3.10 yaml-0.1 --cflags)"
+        - "CFLAGS": "$(pkg-config python-3.12 yaml-0.1 --cflags)"
         - "PIP_NO_BINARY": ":all:"
+        - UV_FROZEN: "true"
+        - UV_PYTHON: /usr/bin/python3
     organize:
         bin/craftctl: libexec/rockcraft/craftctl
         # Extensions currently expect to find data files in share/rockcraft/
         "**/site-packages/extensions": share/rockcraft/extensions
     override-build: |
+      uv export --no-dev --no-emit-workspace --no-emit-package pywin32 --output-file uv-requirements.txt
```

Comment:
This pywin32 item is interesting. Do you know where it came from?

Reply:
kind of; craft-cli [depends on pywin32](https://github.com/canonical/craft-cli/blob/main/pyproject.toml#L11) when in win32, and somehow `uv export` is adding that dependency even on `!win32`